github/pulls_test.go

```diff
@@ -270,12 +271,12 @@ func TestPullRequestsService_Edit(t *testing.T) {
 			defer teardown()
 
 			mux.HandleFunc("/repos/o/r/pulls/1", func(w http.ResponseWriter, r *http.Request) {
-				v := new(pullRequestUpdate)
-				json.NewDecoder(r.Body).Decode(v)
-
 				testMethod(t, r, "PATCH")
-				if !reflect.DeepEqual(v, tt.wantUpdate) {
-					t.Errorf("Request body = %+v, want %+v", v, tt.wantUpdate)
+
+				gotUpdate, _ := ioutil.ReadAll(r.Body)
+				wantUpdate := tt.wantUpdate + "\n" // json encoder adds a newline
+				if string(gotUpdate) != wantUpdate {
+					t.Errorf("Request body = %q, want %q", gotUpdate, wantUpdate)
 				}
```

Comment:
You can use [`testBody`](https://github.com/google/go-github/blob/c9c37fd6ce0247943d1aa34fc3813c62c5cb0225/github/github_test.go#L117-L125) helper to simplify this.

See https://github.com/google/go-github/blob/c9c37fd6ce0247943d1aa34fc3813c62c5cb0225/github/pulls_test.go#L478 for example.